ecosystem/indexer-grpc/indexer-grpc-cache-worker/src/worker.rs

```diff
@@ -345,93 +359,96 @@ async fn process_streaming_response(
-
-                    PROCESSED_VERSIONS_COUNT.inc_by(num_of_transactions);
-                    // TODO: Reasses whether this metric useful
-                    LATEST_PROCESSED_VERSION_OLD.set(current_version as i64);
-                    PROCESSED_BATCH_SIZE.set(num_of_transactions as i64);
-                    info!(
-                        start_version = start_version,
-                        num_of_transactions = num_of_transactions,
-                        "[Indexer Cache] Data chunk received.",
-                    );
-                },
-                GrpcDataStatus::StreamInit(new_version) => {
-                    error!(
-                        current_version = new_version,
-                        "[Indexer Cache] Init signal received twice."
-                    );
-                    ERROR_COUNT.with_label_values(&["data_init_twice"]).inc();
-                    break;
-                },
-                GrpcDataStatus::BatchEnd {
-                    start_version,
-                    num_of_transactions,
-                } => {
-                    info!(
-                        start_version = start_version,
-                        num_of_transactions = num_of_transactions,
-                        "[Indexer Cache] End signal received for current batch.",
-                    );
-                    if current_version != start_version + num_of_transactions {
-                        error!(
-                            current_version = current_version,
-                            actual_current_version = start_version + num_of_transactions,
-                            "[Indexer Cache] End signal received with wrong version."
-                        );
-                        ERROR_COUNT
-                            .with_label_values(&["data_end_wrong_version"])
-                            .inc();
-                        break;
-                    }
-                    cache_operator
-                        .update_cache_latest_version(transaction_count, current_version)
+        match received.response.unwrap() {
+            Response::Status(status) => {
+                match StatusType::try_from(status.r#type)
+                    .expect("[Indexer Cache] Invalid status type.")
+                {
+                    StatusType::Init => panic!("Init signal received twice."),
+                    StatusType::BatchEnd => {
+                        let start_version = status.start_version;
+                        let num_of_transactions = status
+                            .end_version
+                            .expect("TransactionsFromNodeResponse status end_version is None")
+                            + 1
+                            - start_version;
+                        tx.send(CacheTaskItem::UpdateCacheLatestVersion((
+                            num_of_transactions,
+                            start_version,
+                        )))
```

Comment:
why do we need a separate item for this? this makes it more complicated than it needs to be. why not spin up 10 threads, process batch of size X, check for gaps, then update cache latest version. similar to how it's done in the filestore worker now.